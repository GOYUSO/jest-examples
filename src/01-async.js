export async function fetchData(id) {
  if (!id) {
    throw new Error("ID is required");
  }

  // Simulate an API call
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve({ id, name: "John Doe" });
    }, 500);
  });
}
