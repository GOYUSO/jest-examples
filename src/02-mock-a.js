export async function fetchData(id) {
  return { id, name: "John Doe" };
}
