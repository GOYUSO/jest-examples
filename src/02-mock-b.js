export function sendMessage(service, message) {
  return service.deliver(message);
}
