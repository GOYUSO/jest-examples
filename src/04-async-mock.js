export async function fetchUserData(userId) {
  const response = await fetch(
    `https://jsonplaceholder.typicode.com/users/${userId}`
  );
  if (!response.ok) {
    throw new Error("Failed to fetch user data");
  }
  return response.json();
}
