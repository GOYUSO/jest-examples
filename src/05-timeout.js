export async function retryOperation(operation, retries, delay) {
  for (let attempt = 1; attempt <= retries; attempt++) {
    try {
      return await operation();
    } catch (error) {
      if (attempt < retries) {
        await new Promise((resolve) => setTimeout(resolve, delay));
      } else {
        throw new Error(`Operation failed after ${retries} retries`);
      }
    }
  }
}
