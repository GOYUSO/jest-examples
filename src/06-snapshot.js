export function getUserProfile() {
  return {
    id: 1,
    name: "John Doe",
    email: "john.doe@example.com",
    isActive: true,
    createdAt: new Date(2024, 11, 24).toISOString(),
  };
}
