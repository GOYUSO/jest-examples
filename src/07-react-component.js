import React, { useState } from "react";

function Greeting({ name }) {
  const [message, setMessage] = useState("Hello");

  const handleClick = () => {
    setMessage("Goodbye");
  };

  return (
    <div>
      <h1>
        {message}, {name}!
      </h1>
      <button onClick={handleClick}>Change Message</button>
    </div>
  );
}

export default Greeting;
