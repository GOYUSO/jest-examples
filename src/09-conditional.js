import React from "react";

function WelcomeMessage({ isLoggedIn }) {
  const message = isLoggedIn
    ? "Welcome back, user!"
    : "Please log in to continue.";
  return <h1>{message}</h1>;
}

export default WelcomeMessage;
