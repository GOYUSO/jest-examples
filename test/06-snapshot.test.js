import { getUserProfile } from "../src/06-snapshot";

test("user profile matches the snapshot", () => {
  const userProfile = getUserProfile();
  expect(userProfile).toMatchSnapshot();
});
