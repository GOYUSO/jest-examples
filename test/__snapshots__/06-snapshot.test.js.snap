// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile matches the snapshot 1`] = `
{
  "createdAt": "2024-12-23T23:00:00.000Z",
  "email": "john.doe@example.com",
  "id": 1,
  "isActive": true,
  "name": "John Doe",
}
`;
